src/value.h

```diff
@@ -150,6 +150,10 @@ class Tensor: public ShapedValue {
 
   size_t getRank() const { return dims.size(); }
 
+  // Return <a new tensor T2, inbounds>
+  // T2[idx] = idx == indices ? value : this[idx]
+  std::pair<Tensor, smt::Expr> insert(const smt::Expr &value, const std::vector<smt::Expr> &indices) const;
+
```

Comment:
This piece of code apparently came along while merging the branch with master :o